kura/org.eclipse.kura.driver.opcua.provider/src/main/java/org/eclipse/kura/internal/driver/opcua/DataTypeMapper.java

```diff
@@ -43,72 +43,121 @@ private static Variant mapByteArray(byte[] javaValue, VariableType opcuaType) {
+            } else if (targetType == VariableType.DOUBLE) {
                 return new Variant(numberValue.doubleValue());
-            } else if (opcuaType == VariableType.STRING) {
+            } else if (targetType == VariableType.STRING) {
                 return new Variant(numberValue.toString());
             }
         }
-        if (javaValue instanceof String) {
-            final String stringValue = (String) javaValue;
-            if (opcuaType == VariableType.SBYTE) {
+        if (value instanceof String) {
+            final String stringValue = (String) value;
+            if (targetType == VariableType.SBYTE) {
                 return new Variant(Byte.parseByte(stringValue));
-            } else if (opcuaType == VariableType.INT16) {
+            } else if (targetType == VariableType.INT16) {
                 return new Variant(Short.parseShort(stringValue));
-            } else if (opcuaType == VariableType.INT32) {
+            } else if (targetType == VariableType.INT32) {
                 return new Variant(Integer.parseInt(stringValue));
-            } else if (opcuaType == VariableType.INT64) {
+            } else if (targetType == VariableType.INT64) {
                 return new Variant(Long.parseLong(stringValue));
-            } else if (opcuaType == VariableType.BYTE) {
+            } else if (targetType == VariableType.BYTE) {
                 return new Variant(UByte.valueOf(stringValue));
-            } else if (opcuaType == VariableType.UINT16) {
+            } else if (targetType == VariableType.UINT16) {
                 return new Variant(UShort.valueOf(stringValue));
-            } else if (opcuaType == VariableType.UINT32) {
+            } else if (targetType == VariableType.UINT32) {
                 return new Variant(UInteger.valueOf(stringValue));
-            } else if (opcuaType == VariableType.UINT64) {
+            } else if (targetType == VariableType.UINT64) {
                 return new Variant(ULong.valueOf(stringValue));
-            } else if (opcuaType == VariableType.FLOAT) {
+            } else if (targetType == VariableType.FLOAT) {
                 return new Variant(Float.parseFloat(stringValue));
-            } else if (opcuaType == VariableType.DOUBLE) {
+            } else if (targetType == VariableType.DOUBLE) {
                 return new Variant(Double.parseDouble(stringValue));
-            } else if (opcuaType == VariableType.STRING) {
+            } else if (targetType == VariableType.STRING) {
                 return new Variant(stringValue);
             }
         }
-        throw new IllegalArgumentException(
-                message.errorValueTypeConversion() + javaValue.getClass() + " " + opcuaType.name());
+        throw new IllegalArgumentException("Error while converting the retrieved value to the defined typed "
+                + value.getClass() + " " + targetType.name());
+    }
+
+    private static byte[] toByteArray(Object calue) {
```

Comment:
value?

Reply:
I saw it too, but there's `value` a couple lines down as well... Would be interested to know what `calue` stands for.